api/blueprints/drbCollection.py

```diff
@@ -448,11 +491,35 @@ def _buildPublications(editions):
 
     return opdsPubs
 
-def removeEditionsFromCollection(dbClient, collection):
+def removeEditionsFromCollection(dbClient, collection, editionIDs=None, workUUIDs=None):
 
     '''Deleting the rows of collection_editions that were in the original collection'''
 
-    dbClient.session.execute(COLLECTION_EDITIONS.delete().where(COLLECTION_EDITIONS.c.collection_id == collection.id))
+    #Delete every row of the collection
+    if editionIDs == None and workUUIDs == None:
+        dbClient.session.execute(COLLECTION_EDITIONS.delete()\
+            .where(COLLECTION_EDITIONS.c.collection_id == collection.id))
```

Comment:
This behavior strikes me as a *little* unexpected. If I pass `None` to a method to remove things, I would probably expect nothing happen to happen.  Maybe this part should be a separate function, like `removeAllEditionsFromCollection`?   Thoughts?

That said, it looks like the calling code as written will never reach this case.

Reply:
This case will be called for the collectionReplace endpoint specifically because in that endpoint there is no editionIDs or workUUIDs passed as parameters. I'll take you advice and put this one edge case into into its own method though.